modules/rostests/apitests/user32/ScrollBarRedraw.c

```diff
@@ -0,0 +1,727 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
```

Comment:
Are we sure we want to license this under MIT? Maybe GPL is better? :thinking: Looking to all other files even in this apitest, they are under GPL. Or should it depend purely on the author's desidion? If so, and we can't change it against the author's choice, then OK. But if we can, then GPL seems better for me, since this code is a part of ReactOS and it's not even a common stuff like e. g. declarations in public ddk/ndk/psdk/xdk headers.

Reply:
I always prefer MIT instead of GPL.
GPL is like covid, it is infectious in a bad way.

Reply:
OK, maybe it's an argument in general. But as we know, the code licensed under MIT can be taken, used and even closed by anybody. Isn't that a significant disadvantage? Moreover, this one single part of our apitests can't be useful as-is like that, because all others are licensed under GPL. I suspect it will even not compile, because it depends on other GPL'ed files. So does it make any other sense in this specific case still?

Reply:
My reasoning is: Am I willing to go to court to enforce the GPL if we were to find someone or some organization using my code in a manner that violates the GPL?  And would I honestly care?  The answer to both of those questions is no.  I follow the copy-free spirit of the BSDs, anyone should be able to use the code for any purpose they desire, as long as they give credit.

And as for compiling the test on its own, outside of ReactOS, it's as simple as replacing the headers with the following:
```
#include <assert.h>
#include <windows.h>
#include "wine/test.h"
```
and using the test in the standard Wine test suite, or you can very easily roll your own test.h to compile the test standalone.

I don't think the test being licensed under a permissive license should cause any issues.  You're free to use it, and linking it with GPL code is okay - the source code is MIT, and the resulting binary is GPL'd.  If, however, you insist on the code of the test being released under the GPL as well, in order for it to be effective, we would have to arrange the transferal of copyright ownership to the ReactOS project, as you cannot expect me to personally go into court to "fight for this testcase".

Reply:
OK, if so, then keep it under MIT. Your arguments are right.

> And as for compiling the test on its own, outside of ReactOS, it's as simple as replacing the headers with the following:

> #include <assert.h>
#include <windows.h>
#include "wine/test.h"

Indeed. Then maybe it makes sense to add these includes into `#ifdnef __REACTOS__` or some another appropriate define, to make it compilable outside of ReactOS, and place ReactOS required defines into `#else` clause then? Or on the contrary: place into `#ifdef __REACTOS__` ROS includes, and then non-ROS ones into `#else` clause? And at the final, add the note about them: which includes are corresponding to compile in ROS environment and which are not. :thinking:

Reply:
I don't think adding those ifdefs is necessary.  If someone wants to use the testcase outside of ReactOS, they can put in the effort to replace the user32 apitest include (precomp.h) with whatever is necessary for running it outside of this bundle of tests.  If someone from the Wine project wanted to copy it, they'd very quickly know what they need to change in order to get it running after giving the code a quick glance.

Reply:
OK, then it's not necessarily.

Reply:
Worst case scenario, they can always hit me up with an email :)  The address is in the copyright notice.  But yeah, I'm glad we were able to reach a consensus!